src/backend/crew_ai/callbacks.py

```
@@ -0,0 +1,133 @@
+
+def make_step_callback(step_logger: logging.Logger):
+    """Return a callback that logs a timestamped entry per agent step.
+
+    Fires once per thought/action/observation cycle. Each entry includes
+    elapsed time from the previous step so you can measure per-LLM-call
+    latency by diffing consecutive lines.
+
+    Handles both AgentAction-style objects (newer CrewAI) and plain strings
+    (older versions) via graceful attribute access.
+    """
+    state = {"last_ts": datetime.now()}
```

Ruff (0.15.7)

[warning] `datetime.datetime.now()` called without a `tz` argument

(DTZ005)